src/interpreters/Neorg_original.rs

```diff
@@ -0,0 +1,329 @@
+#[derive(Clone)]
+#[allow(non_camel_case_types)]
+pub struct Neorg_original {
+    support_level: SupportLevel,
+    data: DataHolder,
+    code: String,
+
+    default_filetype: String,
+}
+
+impl Neorg_original {
+    pub fn get_filetype_of_embbeded_code(&mut self) -> Result<String, SniprunError> {
+        let nvim_instance = self.data.nvim_instance.clone().unwrap();
+        let mut real_nvim_instance = nvim_instance.lock().unwrap();
+
+        // walk the whole visual selection in case multiple code block are contained
+        let lines = real_nvim_instance
+            .get_current_buf()
+            .unwrap()
+            .get_lines(
+                &mut real_nvim_instance,
+                self.data.range[0] - 1,
+                self.data.range[1],
+                false,
+            )
+            .unwrap();
+        let mut counter = 0;
+        let selection_line = self.data.range[0] as usize;
+        let mut v = vec![];
+        for (i, l) in lines.iter().enumerate() {
+            info!("checking code bloc delimiter in : {}",l);
+            if l.trim_start().to_lowercase().starts_with("@code") {
+                if counter % 2 == 1 { return Err(SniprunError::CustomError(String::from("Incomplete or nested code blocs")))} 
+                counter += 1;
+                v.push((selection_line + i + 1, 0));
+            }
+            if l.trim_start().to_lowercase().starts_with("@end") {
+                if counter % 2 == 0 { return Err(SniprunError::CustomError(String::from("Incomplete or nested code blocs")))} 
+                counter += 1;
+                v[((counter - 1) / 2) as usize].1 = selection_line + i - 1;
+            }
+        }
+        if counter >= 2 {
+            info!("counting {} code blocs delimiters", counter);
+            if counter % 2 == 1 {
+                return Err(SniprunError::CustomError(String::from(
+                    "Selection contains an odd number of code bloc delimiters",
+                )));
+            }
+            info!("running separately ranges : {:?}", v);
+            return Err(SniprunError::ReRunRanges(v));
+        }
+        info!("no multiple bloc was found");
```

Comment:
here is where you want to call return ReRunRange(your_code_bloc_range) to have the virtual text result displayed at the _end_ of the code bloc. Beware of not  re-returning ReRunRange the second time this code is called ! 

(you can store/read data persistently (between runs) with save_code() and read_previous_code()